src/serverrpc.cpp

```diff
@@ -0,0 +1,238 @@
+/******************************************************************************\
+ * Copyright (c) 2021
+ *
+ * Author(s):
+ *  dtinth
+ *
+ ******************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ *
+ \******************************************************************************/
+
+#include "serverrpc.h"
+
+CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* parent ) : QObject ( parent )
+{
+    // API doc already part of CClientRpc
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "server" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    /// @rpc_method jamulusserver/getRecorderStatus
+    /// @brief Returns the recorder state.
+    /// @param {object} params - No parameters (empty object).
+    /// @result {boolean} result.initialised - True if the recorder is initialised.
+    /// @result {string} result.errorMessage - The recorder error message, if any.
+    /// @result {boolean} result.enabled - True if the recorder is enabled.
+    /// @result {string} result.recordingDirectory - The recorder recording directory.
+    pRpcServer->HandleMethod ( "jamulusserver/getRecorderStatus", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{
+            { "initialised", pServer->GetRecorderInitialised() },
+            { "errorMessage", pServer->GetRecorderErrMsg() },
+            { "enabled", pServer->GetRecordingEnabled() },
+            { "recordingDirectory", pServer->GetRecordingDir() },
+        };
+
+        response["result"] = result;
```

Comment:
This used to be nested one level deeper (i.e. `result.recorder.enabled`) while the method was still called `getServerStatus`. I've moved it up one level as I think the extra distinction is now obsolete. Pointing this out explicitly nevertheless.